app/src/main/java/com/infomaniak/mail/ui/main/newmessage/NewMessageViewModel.kt

```diff
@@ -20,62 +20,97 @@ package com.infomaniak.mail.ui.main.newmessage
+        viewModelScope.launch(Dispatchers.IO) {
+            val draft = draftResources?.let { MailApi.fetchDraft(it, messageUid ?: "") }
+                ?: draftUuid?.let { MailboxContentController.getDraft(it) }
+                ?: Draft().apply { initLocalValues() }
+            currentDraft.postValue(draft)
+        }
+    }
 
     fun getAllContacts(): List<UiContact> {
         val contacts = mutableListOf<UiContact>()
         MailData.contactsFlow.value?.forEach { contact ->
             contacts.addAll(contact.emails.map { email -> UiContact(email, contact.name) })
         }
+
         return contacts
     }
 
-    fun sendMail(draft: Draft, action: Draft.DraftAction) {
+    fun sendMail(draft: Draft) {
         val mailbox = MailData.currentMailboxFlow.value ?: return
-        fun sendDraft() = ApiRepository.sendDraft(mailbox.uuid, draft.fillForApi("send"))
-        fun saveDraft() = ApiRepository.saveDraft(mailbox.uuid, draft.fillForApi("save"))
-
-        viewModelScope.launch(Dispatchers.IO) {
-            val signature = ApiRepository.getSignatures(mailbox.hostingId, mailbox.mailbox)
-            draft.identityId = signature.data?.defaultSignatureId
-            if (action == Draft.DraftAction.SEND) sendDraft() else saveDraft()
+        val draftWithSignature = if (draft.identityId == null) MailData.setDraftSignature(draft) else draft
+        // TODO: better handling of api response
+        if (draftWithSignature.action == DraftAction.SEND.apiName) {
+            MailData.sendDraft(draftWithSignature, mailbox.uuid)
+        } else {
+            MailData.saveDraft(draftWithSignature, mailbox.uuid).data?.let {
+                currentDraft.value?.apply {
+                    uuid = it.uuid
+                    parentMessageUid = it.uid
+                    isOffline = false
+                }
+            }
         }
     }
 
-    private fun Draft.fillForApi(draftAction: String) = apply {
-        action = draftAction
-        to = recipients.toRealmRecipients() ?: realmListOf()
-        cc = newMessageCc.toRealmRecipients()
-        bcc = newMessageBcc.toRealmRecipients()
+    fun clearJobs() {
+        autoSaveJob?.cancel()
+    }
 
-//        // TODO: manage advanced functionalities
-//        quote = ""
-//        references = ""
-//        delay = 0
-//        inReplyTo = ""
-//        inReplyToUid = ""
-//        replyTo = realmListOf()
+    fun Draft.fill(draftAction: DraftAction, messageEmail: String, messageSubject: String, messageBody: String) {
+        // TODO: Should userInformation (here 'from') be stored in mainViewModel? See ApiRepository.getUser()
+        MailboxContentController.updateDraft(this) {
+            it.from = realmListOf(Recipient().apply { email = messageEmail })
+            it.subject = messageSubject
+            it.body = messageBody
+            it.priority = MessagePriority.Priority.NORMAL.getPriority()
```

Comment:
The name of this method is too misleading, one would think it returns an object when it's just a String, in this case it's better to just modify the `toString` method in the enum directly

Reply:
done